packages/app/src/composables/useLogin.ts

```diff
@@ -16,26 +16,37 @@ type UseLogin = ToRefs<LoginState> & {
   login: () => Promise<void>;
   logout: () => Promise<void>;
   initializeLogin: () => Promise<void>;
+  handlePrividiumCallback: () => Promise<void>;
 };
 
 const state = reactive<LoginState>({
   isLoginPending: false,
 });
 
+let prividiumAuth: PrividiumAuth | null = null;
+
 export default (context: Context, _logger = defaultLogger): UseLogin => {
-  const wallet = useWallet({
-    ...context,
-    currentNetwork: computed(() => ({
-      explorerUrl: context.currentNetwork.value.rpcUrl,
-      chainName: context.currentNetwork.value.l2NetworkName,
-      l1ChainId: 0,
-      ...context.currentNetwork.value,
-    })),
-  });
+  const router = useRouter();
+
+  const getPrividiumAuth = () => {
+    if (!context.currentNetwork.value.prividium) {
+      return null;
+    }
```

Comment:
Does it make sense to throw an Exception here instead of returning a `null`?
```
    if (!context.currentNetwork.value.prividium) {
      throw new Error("Prividium authentication is not configured");
    }
```
It's thrown up the callstack anyway. This way it will be in one place.